### PR TITLE
certs: fix SKID/AKID generation

### PR DESCRIPTION
**Pull request by stlaz**
The leaf certificate Subject Key Identifier generation was broken as it was assigning the CA pub key fingerprint to the leaf certificate SKID.

This commit removes the logic completely, golang generates the SKIDs/AKIDs where necessary.

/assign @csrwng 
/cc @ibihim 

---------

related rfc: https://www.rfc-editor.org/rfc/rfc3280#section-4.2.1.2

### CODE REVIEW

File: support/certs/tls.go [hunk 1]
```diff
@@ -166,39 +151,14 @@ func signedCertificate(
 		Version:               3,
 		BasicConstraintsValid: true,
 	}
-	pub := caCert.PublicKey.(*rsa.PublicKey)
-	certTmpl.SubjectKeyId, err = generateSubjectKeyID(pub)
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to set subject key identifier")
-	}
+
```
Comment (ibihim) [inline support/certs/tls.go:154]: An alternative solution that would keep SKID would be

```go
h := sha1.Sum(key.Public())
certTmpl.SubjectKeyId = h[:]
```

right?

File: support/certs/tls.go [hunk 2]
```diff
@@ -166,37 +157,25 @@ func signedCertificate(
 		Version:               3,
 		BasicConstraintsValid: true,
 	}
-	pub := caCert.PublicKey.(*rsa.PublicKey)
-	certTmpl.SubjectKeyId, err = generateSubjectKeyID(pub)
+
+	certTmpl.SubjectKeyId, err = rsaPubKeySHA1Hash(&key.PublicKey)
```
Comment (ibihim) [inline support/certs/tls.go:161]: @stlaz, we only use RSA?

Comment (stlaz) [reply support/certs/tls.go:161]: the private keys on the input of both the functions are RSA so it's safe to assume this

**Pull request merged**